kubetools/kubernetes/api.py

```diff
@@ -466,3 +309,100 @@ def check_job():
             return True
 
     _wait_for(check_job, get_object_name(job))
+
+
+def _wait_for_object(*args):
+    return _wait_for(lambda: _object_exists(*args) is True)
+
+
+def _wait_for_no_object(*args):
+    return _wait_for(lambda: _object_exists(*args) is False)
+
+
+def _wait_for(function, name='object'):
+    settings = get_settings()
+
+    sleeps = 0
+    while True:
+        if function():
+            return
+
+        sleep(settings.WAIT_SLEEP_TIME)
+        sleeps += 1
+
+        if sleeps > settings.WAIT_MAX_SLEEPS:
+            raise KubeBuildError(f'Timeout waiting for {name} to be ready')
+
+
+def _object_exists(api, method, namespace, obj):
+    try:
+        if namespace:
+            getattr(api, method)(
+                namespace=namespace,
+                name=get_object_name(obj),
+            )
+        else:
+            getattr(api, method)(
+                name=get_object_name(obj),
+            )
+    except ApiException as e:
+        if e.status == 404:
+            return False
+        raise
+    return True
+
+
+def _get_cronjob_api_version(cronjob_obj):
+    default_version = get_settings().CRONJOBS_BATCH_API_VERSION
+    return cronjob_obj.get('apiVersion', default_version)
+
+
+def _get_compatible_cronjob_api(env, default_version=None):
+    possible_versions = [
+        'batch/v1',
+        'batch/v1beta1',
+    ]
+    if default_version:
+        possible_versions.insert(0, default_version)
+
+    for version in possible_versions:
+        batch_api = _get_k8s_cronjobs_batch_api(env, version)
+        api_resources = batch_api.get_api_resources()
```

Comment:
Assume you've tested this, but does this raise an error when the cluster does not support a version. For example, when using an older k8s version that doesn't support `batch/v1`, does `get_api_resources()` return an empty list or raise an exception?

Reply:
I've tested this with k8s versions down to 1.18 and encountered no issue

Reply:
100644
Added that in the release PR